torch/amp/autocast_mode.py

```diff
@@ -258,6 +270,12 @@ def __enter__(self):
             torch.hpu.set_autocast_hpu_enabled(self._enabled)  # type: ignore[attr-defined]
             torch.hpu.set_autocast_hpu_dtype(self.fast_dtype)  # type: ignore[attr-defined]
             torch.autocast_increment_nesting()
+        elif self.device == self.custom_backend_name:
+            self.prev = eval('torch.{0}.is_autocast_{0}_enabled()'.format(self.custom_backend_name))
+            self.prev_fastdtype = eval('torch.{0}.get_autocast_{0}_dtype()'.format(self.custom_backend_name))
+            eval('torch.{0}.set_autocast_{0}_enabled(self._enabled)'.format(self.custom_backend_name))
+            eval('torch.{0}.set_autocast_{0}_dtype(self.fast_dtype)'.format(self.custom_backend_name))
```

Comment:
Instead of using eval() here, can we use getattr/hasattr?

`assert hasattr(torch, self.custom_backend_name), "Tried to use AMP with the {___} backend, but the backend has not registered a module. The backend should register a corresponding module with `torch._register_device_module`" (add more to the error message)

`custom_device_mod = getattr(torch, self.custom_backend_name)`
** do stuff with custom device module **